scanner/go.go

```diff
@@ -1,19 +1,87 @@
 package scanner
 
+import (
+	"golang.org/x/mod/modfile"
+	"os"
+	"path"
+)
+
+var gomod *modfile.File
+
 func configureGo(sourceDir string, config *ScannerConfig) (*SourceInfo, error) {
-	if !checksPass(sourceDir, fileExists("go.mod", "Gopkg.lock")) {
+	if !checksPass(sourceDir, fileExists("go.mod")) {
 		return nil, nil
 	}
 
+	modfileParsed := parseModfile()
+
+	version := extractGoVersion(modfileParsed)
+	binaryName := generateBinaryName(modfileParsed)
+
+	files := templates("templates/go")
+
 	s := &SourceInfo{
-		Builder:    "paketobuildpacks/builder:base",
-		Buildpacks: []string{"gcr.io/paketo-buildpacks/go"},
-		Family:     "Go",
-		Port:       8080,
+		Files:  files,
+		Family: "Go",
+		Port:   8080,
 		Env: map[string]string{
 			"PORT": "8080",
 		},
+		BuildArgs: map[string]string{
+			"GO_VERSION":  version,
+			"GO_BIN_NAME": binaryName,
+		},
 	}
 
 	return s, nil
 }
+
+func parseModfile() bool {
+	dat, err := os.ReadFile("go.mod")
+	if err != nil {
+		return false
+	}
+
+	f, modErr := modfile.Parse("go.mod", dat, nil)
+
+	if modErr != nil {
+		return false
+	}
+
+	gomod = f
+
+	return true
+}
+
+func extractGoVersion(modfileParsed bool) string {
+	// todo: Can we get current latest stable via web request?
```

Comment:
what about using `"1"` which according to [golang offiicial image](https://hub.docker.com/_/golang) always point to latest Go v1 release.